src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -584,6 +584,8 @@ pass::EliminateConcatStridedSlice::EliminateConcatStridedSlice() {
                 if (end_constant_node == nullptr)
                     return false;
                 auto end_values = end_constant_node->cast_vector<int64_t>();
+                if (end_values[concat_axis] == INT64_MAX)
```

Comment:
```suggestion
                if (end_values[concat_axis] == std::numeric_limits<int64_t>::max())
```

Add tests, which reproduce reported issue, please.